documentation/specs/src/economics/public-goods-funding.md

```diff
@@ -38,8 +28,237 @@ Namada groups public goods into four categories, with earmarked pools of funding
+
+The collecting member of the council will then be responsible for submitting this tx through the multisig. The multisig will only accept the tx if this is true.
+
+Note that there is no cap to $n$ in $k$-out-of-$n$ apart from the limitations of the multisig account.
+
+The PGF council should be able to make both retroactive and continuous public funding transactions. Retroactive public funding transactions should be straightforward and implement no additional logic to a normal transfer.
+
+However, for continuous PGF (cPGF), the council should be able to submit a one time transaction which indicates the recipient addresses that should be eligble for receiveing cPGF. 
+
+The following data is attached to the PGF transaction and will allow the counsil to decide which projects will be continously funded. Each tuple represent the address and the respecting amount of NAM that will receive every epoch. The list of funding recipients will be stored in storage under the PGF internal address substorage space.
+
+```rust
+struct cPgfRecipients {
+    recipients: HashSet<(Address, u64)>
+}
+```
+
+The mechanism for these transfers will be implemented in `finalize-block.rs`, which will send the addresses their respective amounts each end-of-epoch.
+
+## Addresses
+Governance adds 1 internal address:
+
+`PGF` internal address
+
+The internal address VP will hold the allowance the 10% inflation of NAM. This will be added in addition to what was unspent by the previous council. It is important to note that it is this internal address which holds the funds, rather than the PGF council multisig.
+
+The council should be able to burn funds (up to their spending cap), but this hopefully should not require additional functionality beyond what currently exists.
+
+Further, the VP should contain the parameter that dictates the number of epochs a candidacy is valid for once it has been broadcasted and before it needs to be renewed.
+
+### VP checks
+
+The VP must check that the council does not exceed its spending cap.
+
+The VP must also check that the any spending is only done by a the correctly elected PGF council multisig address.
+
+## Storage
+
+### Storage keys
+
+Each recipient will be listed under this storage space (for cPGF)
+- `/PGFAddress/cPGF_recipients/Address = Amount`
+- `/PGFAddress/spending_cap = Amount`
+- `/PGFAddress/spent_amount = Amount`
+- `/PGFAddress/candidacy_length = u8`
+- `/pgf/counsil_candidates/epoch = HashSet<Council>`
+- `/PGFAddress/council/epoch = Council` (The council that was elected at epoch `epoch`)
+
+### Struct
+
+```rust
+struct Council {
+    address: Address,
+    spending_cap_amount: u64,
+    spent_amount: u64,
+}
+```
+The spending cap amount here is different from the `spending_cap` float. This is an actual NAM value that is calculated through `spending_cap * PGF_VP.balance`
```

Comment:
float? where's a float? there should never be any floats anywhere ever :smile:

Reply:
also I don't understand what this is - why would we multiply the spending cap by the balance?

Reply:
Well I thought of spending_cap as a percentage of the balance. I guess we can make it a uint as well and it directly be the literal amount of NAM. Benefit of percentage is that it is interpretable and easy to remember whereas Amount is not (could be many zeros).

Reply:
You can use the `Amount` type for the spending cap

Reply:
I just don't think `Amount` is interpretable and can lead to a lot of misvotes. I'd rather use a percentage for spending cap.